src/attribute-converters.ts

```diff
@@ -0,0 +1,82 @@
+import { PropValue } from './types';
+
+const parseOrUndefined = (value: string): object => {
+  try {
+    return JSON.parse(value);
+  } catch (_) {
+    return undefined;
+  }
+};
+
+export const toBoolean = (prop?: PropValue): boolean => {
+  if (typeof prop === 'boolean') {
+    return prop;
+  }
+  return (!!prop && prop !== 'false') || prop === '';
+};
+
+export const toNumber = (prop?: PropValue): number => {
+  if (typeof prop === 'number') {
+    return prop;
+  }
+  let convertedProp = +prop;
+  // eslint-disable-next-line no-self-compare
+  if (convertedProp !== convertedProp) { // if (is NaN)
+    const float = parseFloat(prop as string);
+    convertedProp = float || float === 0 ? float : convertedProp;
+  }
+  return convertedProp;
+};
+
+export const toArray = <T>(prop?: PropValue): T[] => {
+  if (Array.isArray(prop)) {
+    return prop;
+  }
+  if (typeof prop === 'string') {
+    const convertedProp = parseOrUndefined(prop);
+    return typeof convertedProp !== 'object' ? null : Object.keys(convertedProp)
+      .reduce(<T>(accumulator: T[], key: string): T[] => ([
+        ...accumulator,
+        convertedProp[key],
+      ]), []);
+  }
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return [...(new Array(prop))] as any[];
+};
+
+export const toObject = (prop?: PropValue): object => {
+  if (typeof prop === 'object' && !Array.isArray(prop)) {
+    return prop;
+  }
+  if (typeof prop === 'string') {
+    const convertedProp = parseOrUndefined(prop);
+    return typeof convertedProp !== 'object' ? null : Object.keys(convertedProp)
+      .reduce((accumulator: object, key: string): object => ({
+        ...accumulator,
+        [key]: convertedProp[key],
+      }), {});
+  }
+  return [...(new Array(prop))].reduce((acc, value, index) => ({ ...acc, [index]: value }), {});
+};
+
+export const toFunction = (prop?: PropValue): Function => {
+  if (typeof prop === 'function') {
+    return prop;
+  }
+  try {
+    // eslint-disable-next-line no-new-func
+    return (new Function(`return ${prop};`))();
+  } catch (_) {
+    return null;
```

Comment:
That's a silent error. Can we throw an error in the console? And as soon as we have a development bundle and a production bundle we can change the behaviour to throw in development?

Reply:
Here I disagree. The converters should do their best to convert to that specific type. It just so happens that `new Function` sometimes throws an error. But that's a detail of the implementation itself. IMO the output of the converters should always be "the type the dev requested" or "null" - nothing else (error throwing included).

Reply:
Fair point, let's see how it goes and add a warning to the docs if needed.